src/coreComponents/constitutive/solid/TriaxialDriver.hpp

```diff
@@ -0,0 +1,205 @@
+ *
+ */
+class TriaxialDriver : public TaskBase
+{
+public:
+  TriaxialDriver( const string & name,
+                  Group * const parent );
+  ~TriaxialDriver() override;
+
+  static string catalogName() { return "TriaxialDriver"; }
+
+  void postProcessInput() override;
+
+  virtual bool execute( real64 const GEOSX_UNUSED_PARAM( time_n ),
+                        real64 const GEOSX_UNUSED_PARAM( dt ),
+                        integer const GEOSX_UNUSED_PARAM( cycleNumber ),
+                        integer const GEOSX_UNUSED_PARAM( eventCounter ),
+                        real64 const GEOSX_UNUSED_PARAM( eventProgress ),
+                        DomainPartition & GEOSX_UNUSED_PARAM( domain ) ) override;
+
+  /**
+   * @brief Run a strain-controlled test using loading protocol in table
+   * @param solid Solid constitutive model
+   * @param table Table with stress / strain time history
+   */
+  template< typename SOLID_TYPE >
+  void runStrainControlTest( SOLID_TYPE & solid, arrayView2d< real64 > & table );
+
+  /**
+   * @brief Run a mixed stress/strain-controlled test using loading protocol in table
+   * @param solid Solid constitutive model
+   * @param table Table with stress / strain time history
+   */
+  template< typename SOLID_TYPE >
+  void runMixedControlTest( SOLID_TYPE & solid, arrayView2d< real64 > & table );
+
+  /**
+   * @brief Ouput table to file for easy plotting
+   */
+  void outputResults();
+
+  /**
+   * @brief Read in a baseline table from file and compare with computed one (for unit testing purposes)
+   */
+  void compareWithBaseline();
+
+private:
+
+  /**
+   * @struct viewKeyStruct holds char strings and viewKeys for fast lookup
+   */
+  struct viewKeyStruct
+  {
+    constexpr static char const * solidMaterialNameString() { return "material"; }
+    constexpr static char const * modeString() { return "mode"; }
+    constexpr static char const * strainFunctionString() { return "strainFunction"; }
+    constexpr static char const * stressFunctionString() { return "stressFunction"; }
+    constexpr static char const * numStepsString() { return "steps"; }
+    constexpr static char const * outputString() { return "output"; }
+    constexpr static char const * baselineString() { return "baseline"; }
+  };
+
+  int m_numSteps;              ///< Number of load steps
+  string m_solidMaterialName;  ///< Material identifier
+  string m_mode;               ///< Test mode: triaxial, volumetric, oedometer
+  string m_strainFunctionName; ///< Time-dependent function controlling strain (role depends on test mode)
+  string m_stressFunctionName; ///< Time-dependent function controlling stress (role depends on test mode)
+  string m_outputFile;         ///< Output file (optional, no output if not specified)
+  Path m_baselineFile;         ///< Baseline file (optional, for unit testing of solid models)
+  array2d< real64 > m_table;   ///< Table storing time-history of axial/radial stresses and strains
+
```

Comment:
///< is a new way for documenting private parameters?

Reply:
This is a standard doxygen syntax for adding @brief documentation after a member is defined. https://www.doxygen.nl/manual/docblocks.html